src/OmniSharp.Roslyn.CSharp/Services/Structure/BlockStructureService.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+using OmniSharp.Mef;
+using OmniSharp.Models;
+using OmniSharp.Services;
+using OmniSharp.Utilities;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Composition;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OmniSharp.Roslyn.CSharp.Services.Structure
+{
+    [OmniSharpHandler(OmniSharpEndpoints.BlockStructure, LanguageNames.CSharp)]
+    public class BlockStructureService : IRequestHandler<Request, IEnumerable<QuickFix>>
```

Comment:
Is `QuickFix` enough? I have two thoughts:

1. VS Code's `FoldingRange` also has a `FoldingRangeKind` that we might want to expose, so we might want a custom model.
2. `QuickFix` is a "kitchen sink" model with lots of baggage that even unnecessarily allocates a `List<string>`. I've never been a fan of overusing it in the protocol. If most of the properties on `QuickFix` aren't interesting, it's probably cleaner to just define a new `BlockStructure` model.